x-pack/plugins/license_api_guard/server/license.ts

```diff
@@ -0,0 +1,115 @@
+  private licenseType?: LicenseType;
+
+  private _isEsSecurityEnabled: boolean = false;
+
+  setup({ pluginName, logger }: SetupSettings) {
+    this.pluginName = pluginName;
+    this.logger = logger;
+  }
+
+  start({ pluginId, minimumLicenseType, licensing }: StartSettings) {
+    licensing.license$.subscribe((license: ILicense) => {
+      this.licenseType = license.type;
+      this.licenseCheckState = license.check(pluginId, minimumLicenseType!).state;
+      // Retrieving security checks the results of GET /_xpack as well as license state,
+      // so we're also checking whether security is disabled in elasticsearch.yml.
+      this._isEsSecurityEnabled = license.getFeature('security').isEnabled;
+    });
+  }
+
+  private getLicenseErrorMessage(licenseCheckState: LicenseCheckState): string {
+    switch (licenseCheckState) {
+      case 'invalid':
+        return i18n.translate('xpack.licenseApiGuard.license.errorUnsupportedMessage', {
+          defaultMessage:
+            'Your {licenseType} license does not support {pluginName}. Please upgrade your license.',
+          values: { licenseType: this.licenseType!, pluginName: this.pluginName },
+        });
+
+      case 'expired':
+        return i18n.translate('xpack.licenseApiGuard.license.errorExpiredMessage', {
+          defaultMessage:
+            'You cannot use {pluginName} because your {licenseType} license has expired.',
+          values: { licenseType: this.licenseType!, pluginName: this.pluginName },
+        });
+
+      case 'unavailable':
+        return i18n.translate('xpack.licenseApiGuard.license.errorUnavailableMessage', {
+          defaultMessage:
+            'You cannot use {pluginName} because license information is not available at this time.',
+          values: { pluginName: this.pluginName },
+        });
+    }
+
+    return i18n.translate('xpack.licenseApiGuard.license.genericErrorMessage', {
+      defaultMessage: 'You cannot use {pluginName} because the license check failed.',
+      values: { pluginName: this.pluginName },
+    });
+  }
+
+  guardApiRoute<Context extends RequestHandlerContext, Params, Query, Body>(
+    handler: RequestHandler<Params, Query, Body, Context>
+  ) {
+    return (
+      ctx: Context,
+      request: KibanaRequest<Params, Query, Body>,
+      response: KibanaResponseFactory
+    ) => {
+      // We'll only surface license errors if users attempt disallowed access to the API.
+      if (this.licenseCheckState !== 'valid') {
+        const licenseErrorMessage = this.getLicenseErrorMessage(this.licenseCheckState);
+        this.logger?.warn(licenseErrorMessage);
+
+        return response.customError({
+          body: {
+            message: licenseErrorMessage,
+          },
+          statusCode: 403,
+        });
+      }
+
+      return handler(ctx, request, response);
+    };
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-member-accessibility
+  get isEsSecurityEnabled() {
```

Comment:
Can we remove the `// eslint-disable`?

```suggestion
  public get isEsSecurityEnabled() {
```